src/main.py

```diff
@@ -1,19 +1,109 @@
 import os
-from telegram.ext import ApplicationBuilder, CommandHandler
+import logging
+from telegram import Update
+from telegram.ext import (
+    ApplicationBuilder,
+    filters,
+    ContextTypes,
+    CommandHandler,
+    MessageHandler,
+)
+
+from classes.chat_registry import ChatRegistry
+from classes.club import Club
+
+REGISTRY_FOLDER = f"{os.getcwd()}/registry"
+
+chat_registry = ChatRegistry(REGISTRY_FOLDER)
+
+
+# setting up logging module https://github.com/python-telegram-bot/python-telegram-bot/wiki/Exceptions%2C-Warnings-and-Logging
+logging.basicConfig(
+    # filename='bot.log',   #TODO add log rotating
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+    level=logging.ERROR,
+)
+
+#  retrieves a logger named after the current module from the logging module's hierarchy of loggers
+logger = logging.getLogger(__name__)
 
 BOT_TOKEN = os.environ.get("BOT_TOKEN")
 
 
-async def start(update, context):
-    await context.bot.send_message(
-        chat_id=update.effective_chat.id, text="Hello, World!"
+# Perhaps not required, can be used as template for team_polling
+async def run_club_polling(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
+    """
+    Runs polling for the club associated with the chat_id.
+    """
+    await chat_registry.run_polling(update.message.chat_id)
+    await chat_registry.delete_message_delayed(
+        update.message.chat_id, update.message.message_id, delay=10
     )
 
 
+# Perhaps not required, can be used as template for team_polling
+# I think it's required to move or guess better add scheduled polling to the team from the club so that there's the option to disable it for certain topics(teams).
+async def stop_club_polling(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
+    """
+    Stops polling for the club associated with the chat_id.
+    """
+    await chat_registry.stop_polling(update.message.chat_id)
+    await chat_registry.delete_message_delayed(
+        update.message.chat_id, update.message.message_id, delay=10
+    )
+
+
+async def start(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
+    """
+    Initiates the start of the bot.
+    Adds the club to the registry.
+    """
+    await chat_registry.add_club(
+        Club(update.message.chat_id, update.effective_chat.title)
+    )
+    await chat_registry.delete_message_delayed(
+        update.message.chat_id, update.message.message_id, delay=10
+    )
+
+
+async def start(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
```

Comment:
looks like a duplication of the above method

Reply:
done